src/server/services/sgid.ts

```diff
@@ -0,0 +1,98 @@
+import SgidClient, { generatePkcePair } from '@opengovsg/sgid-client'
+
+interface SgidServiceOption {
+  clientId: string
+  clientSecret: string
+  privateKey: string
+  redirectUri: string
+  hostname: string
+}
+
+export const SGID_LOGIN_OAUTH_STATE = 'login'
+
+class SgidService {
+  // SGID client will be null for other domains like edu, health
```

Comment:
could you explain what you mean by the sgID Client will be null for other domains?

it is true that there will not be work email present in the `userinfo` requested for end-users logging in without a `.gov.sg` email but how does that make the `sgidClient` null? thanks!

Reply:
Go is deployed to 3 environments, go.gov.sg, for.edu.sg (education link shortener) and for.sg (health link shortener). I believe Prakriti has it set up such that the SGID client is not initialized for the education and health environments!